v2/terraform_flannel.py

```diff
@@ -333,11 +344,12 @@ def collectWindowsLogs(self):
         self.logging.info("Collecting logs.")
         collect_logs_script = os.path.join(os.getcwd(), "collect-logs.ps1")
         try:
+            self.logging.info("Copying collect-logs script to windows nodes.")
+            vms = self.deployer.get_cluster_win_minion_vms_names()
+            self._copyTo(collect_logs_script, "c:\\", vms, windows=True)
+            self._runRemoteCmd(("c:\\collect-logs.ps1 -ArchivePath C:\\k\\logs.zip"), vms, windows=True)
             for vm_name in self.deployer.get_cluster_win_minion_vms_names():
```

Comment:
No reason for the calling self.deployer.get_cluster_win_minion_vm_names again. You already have the list in the vms object.